collector/docker/docker-compose.yml

```diff
@@ -0,0 +1,24 @@
+version: '3.9'
+
+services:
+  otel-collector:
+    container_name: otel
+    image: otel/opentelemetry-collector-contrib
+    environment:
+      LS_ACCESS_TOKEN: ${LS_ACCESS_TOKEN}
+    configs:
+      - source: collector_conf
+        target: /conf/collector.yml
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    user: 10002:997
+    command: ["--config=/conf/collector.yml"]
+    networks:
+      - integrations
```

Comment:
> If Lightstep requires a instance of docker you may be able to use a docker in docker container (runs a docker daemon inside a docker container)
> 
> https://hub.docker.com/_/docker?tab=description

I definitely thought about it. And actually tracked down some useful details about mounting the docker socket by checking articles from the original author of DinD. In this case I felt that it would obscure the heart of the example. More generally, I have run into external networking difficulties with DinD on Mac.